test/xpu_api/numerics/numeric.ops/reduce/xpu_reduce.pass.cpp

```diff
@@ -0,0 +1,108 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// <numeric>
+// UNSUPPORTED: c++03, c++11, c++14
+// UNSUPPORTED: clang-8
+// UNSUPPORTED: gcc-9
+
+// Became constexpr in C++20
+// template<class InputIterator>
+//     typename iterator_traits<InputIterator>::value_type
+//     reduce(InputIterator first, InputIterator last);
+
+#include <oneapi/dpl/numeric>
+#include <cassert>
+#include <CL/sycl.hpp>
+
+#include "support/test_iterators.h"
+
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
+constexpr sycl::access::mode sycl_write = sycl::access::mode::write;
+
+template <class T>
+class KernelTest;
+
+template <typename _T1, typename _T2>
+void
+ASSERT_EQUAL(_T1&& X, _T2&& Y)
```

Comment:
It's very uncommon to name the function with capitals. We may easily end up with conflict with some MACRO name if try to run our tests with some modern framework. I believe I even saw previously such a macro name somewhere.

Reply:
Why do you need forwarding reference? isn't `const` lvalue reference enough?

Reply:
> It's very uncommon to name the function with capitals. We may easily end up with conflict with some MACRO name if try to run our tests with some modern framework. I believe I even saw previously such a macro name somewhere.

Currently, pstl and extension API tests use such code for result check, we follow the same style here.